src/shims/unix/foreign_items.rs

```diff
@@ -315,6 +315,41 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     }
                 }
             }
+            "aligned_alloc" => {
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // The size must be a power of the alignment.
+                // The specs says nothing about error handling other than null
+                // pointer but aligned_alloc Linux/macOs implementations set EINVAL. Note that
+                // on FreeBSD setting a size not being a power of alignment is UB but we do not
+                // want it.
+                //
+                // http://en.cppreference.com/w/cpp/memory/c/aligned_alloc
+                // Linux: https://linux.die.net/man/3/aligned_alloc
+                // FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=aligned_alloc&apropos=0&sektion=3&manpath=FreeBSD+9-current&format=html
```

Comment:
```suggestion
                // The size must be a multiple of the alignment.
                // Alignment must be a power of 2, and "supported by the implementation".
                // If any of these are violated, we have to return NULL.
                //
                // Linux also sets EINVAL, but that's non-standard behavior that we do not emulate.
                // FreeBSD says some of these cases are UB but that's violating the C standard.
                //
                // http://en.cppreference.com/w/cpp/memory/c/aligned_alloc
                // Linux: https://linux.die.net/man/3/aligned_alloc
                // FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=aligned_alloc&apropos=0&sektion=3&manpath=FreeBSD+9-current&format=html
```

Reply:
What about this suggestion? There are still some mistakes in your text, like saying the size must be a "power" of the alignment, which is not correct.